store/storetest/post_store.go

```diff
@@ -898,12 +898,13 @@ func testPostStoreGetPostsBeforeAfter(t *testing.T, ss store.Store) {
 			postList, err := ss.Post().GetPostsAfter(model.GetPostsOptions{ChannelId: channelId, PostId: posts[0].Id, Page: 0, PerPage: -1})
 			assert.Nil(t, postList)
 			assert.Error(t, err)
-			assert.Equal(t, http.StatusBadRequest, err.StatusCode)
+			var invErr *store.ErrInvalidInput
+			assert.True(t, errors.As(err, &invErr))
```

Comment:
Can we assert it with `assert.IsType(t, &store.ErrInvalidInput{}, err)` for example? (The same in the other assertion using this approach).

Reply:
Yeah, great suggestion. I didn't know about that assert method so thanks :+1:

Reply:
Actually, this is incorrect. `IsType` will only check if that variable can be casted to that type or not. Whereas `errors.As` goes down the error chain to check for wrapped errors too. It will work for now, but we are checking for the wrong thing and the code might break in future.